docs/source/cpp/gandiva.rst

```diff
@@ -0,0 +1,151 @@
+.. Licensed to the Apache Software Foundation (ASF) under one
+.. or more contributor license agreements.  See the NOTICE file
+.. distributed with this work for additional information
+.. regarding copyright ownership.  The ASF licenses this file
+.. to you under the Apache License, Version 2.0 (the
+.. "License"); you may not use this file except in compliance
+.. with the License.  You may obtain a copy of the License at
+
+..   http://www.apache.org/licenses/LICENSE-2.0
+
+.. Unless required by applicable law or agreed to in writing,
+.. software distributed under the License is distributed on an
+.. "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+.. KIND, either express or implied.  See the License for the
+.. specific language governing permissions and limitations
+.. under the License.
+
+.. default-domain:: cpp
+.. highlight:: cpp
+.. cpp:namespace:: arrow::compute
+
+===============================
+The Gandiva Expression Compiler
+===============================
+
+Gandiva is a runtime expression compiler that uses `LLVM`_ to generate
+efficient native code for projections and filters on Arrow record batches.
+Gandiva only handles projections and filters. For other transformations, see
```

Comment:
This feels a little awkward after saying it handles projections and filters -- I think you could cut "Gandiva only handles projections and filters," go directly into "For other transformations," and still effectively convey to go to compute functions for all other transformations.

Reply:
There's a lot of concepts flying around in that first sentence though, so I worry that could be a confusing transition. What if I changed "projections and filters" to "compute" in the first instance so it's not redundant?

```suggestion
efficient native code for compute on Arrow record batches.
Gandiva only handles projections and filters; for other transformations, see
```

Reply:
That fixes up the redundancy! I'm happy with that.